app/namespace.go

```diff
@@ -1963,6 +1974,15 @@ func compareCodecSpec(existing, replacement *namespace.CodecServerPropertySpec)
 	return diff.Diff(string(existingBytes), string(replacementBytes)), nil
 }
 
+// Determines if an `AuthMethod` update will result in dropped client connections.
+// Note that `old` and `new` must be different.
+func disruptiveChange(old namespace.AuthMethod, new namespace.AuthMethod) bool {
+	return (new == namespace.AUTH_METHOD_RESTRICTED) ||
```

Comment:
Could this be `old != namespace.AUTH_METHOD_RESTRICTED && new != old`?

Reply:
no, since `(api_key, api_key_or_mtls)` is non-disruptive.

Reply:
but i think `old != namespace.AUTH_METHOD_RESTRICTED && new != namespace.AUTH_METHOD_API_KEY_OR_MTLS` works.